hledger-lib/Hledger/Data/Types.hs

```diff
@@ -237,6 +239,7 @@ type Year = Integer
 data Journal = Journal {
   -- parsing-related data
    jparsedefaultyear      :: (Maybe Year)                          -- ^ the current default year, specified by the most recent Y directive (or current date)
+  ,jparsedefaultmonth     :: (Maybe Month)                         -- ^ the current default month, specified by the most recent Y directive containing month part (or current date)
```

Comment:
Could we disallow having a default month without a default year, perhaps jparsedefaultyear :: Maybe (Year, Maybe Month) ?